### PR TITLE
feat: switch tokio to madsim-tokio

### PR DESCRIPTION
**Pull request by KveinAxel**
this pr should be merge after https://github.com/madsim-rs/madsim/pull/197, and update Cargo.toml

### CODE REVIEW

File: sqlx-core/src/net/socket/mod.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ use crate::io::ReadBuf;
 
 mod buffered;
 
-pub trait Socket: Send + Sync + Unpin + 'static {
+pub trait Socket: Send + Unpin + 'static {
```
Comment (wangrunji0408) [inline sqlx-core/src/net/socket/mod.rs:16]: Could you elaborate on the reason for this change? Are there any madsim structs `!Sync`?

File: sqlx-sqlite/src/lib.rs [hunk 2]
```diff
@@ -106,15 +105,15 @@ pub static CREATE_DB_WAL: AtomicBool = AtomicBool::new(true);
 
 /// UNSTABLE: for use by `sqlite-macros-core` only.
 #[doc(hidden)]
-pub fn describe_blocking(query: &str, database_url: &str) -> Result<Describe<Sqlite>, Error> {
+pub async fn describe_blocking(query: &str, database_url: &str) -> Result<Describe<Sqlite>, Error> {
```
Comment (wangrunji0408) [inline sqlx-sqlite/src/lib.rs:108]: Looks that adding `async` will affect `sqlite-macros` according to the comment. But we are not using the macro, right?

**Review by wangrunji0408 — APPROVED** [review]
I notice that several functions are made into `async` to avoid blocking in simulation. Just want to make sure that:
1. All of them are internal APIs. No public APIs are changed.
2. It won't impact performance in production. (otherwise it'd better to only change it under `#[cfg(madsim)]`)

Right?

File: sqlx-core/src/net/socket/mod.rs (hunk 1 above)
Comment (KveinAxel) [reply sqlx-core/src/net/socket/mod.rs:16]: In https://github.com/madsim-rs/madsim/blob/ca933529dca42bcc5203a98268365fd778a05676/madsim/src/sim/net/mod.rs#L433, madsim use a `futures_util::stream::BoxStream`
And the `BoxStream` doesn't have a type alias with `Sync`

pub type BoxStream<'a, T> = Pin<dyn [Stream<Item = T> + Send + 'a>>;

https://docs.rs/futures-util/latest/futures_util/stream/type.BoxStream.html

**Comment by KveinAxel** [comment]
> I notice that several functions are made into `async` to avoid blocking in simulation. Just want to make sure that:
> 
> 1. All of them are internal APIs. No public APIs are changed.
> 2. It won't impact performance in production. (otherwise it'd better to only change it under `#[cfg(madsim)]`)
> 
> Right?

I'll check it again

File: sqlx-sqlite/src/connection/explain.rs [hunk 3]
```diff
@@ -424,13 +424,14 @@ impl BranchList {
 }
 
 // Opcode Reference: https://sqlite.org/opcode.html
-pub(super) fn explain(
+// TODO: async
```
Comment (fuyufjh) [inline sqlx-sqlite/src/connection/explain.rs:427]:
```suggestion
```

**Pull request merged**